llvm/compiler-course/backend/opolin_d_lab3_var1/ExamplePass.cpp

```diff
@@ -0,0 +1,115 @@
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86Subtarget.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/CodeGen/MachineBasicBlock.h"
+#include "llvm/CodeGen/MachineFunction.h"
+#include "llvm/CodeGen/MachineFunctionPass.h"
+#include "llvm/CodeGen/MachineInstrBuilder.h"
+#include "llvm/CodeGen/MachineRegisterInfo.h"
+
+using namespace llvm;
+
+namespace {
+class AVXLogicCombinerPass : public MachineFunctionPass {
+  const X86InstrInfo *TII = nullptr;
+  MachineRegisterInfo *RegInfo = nullptr;
+  DenseMap<unsigned, unsigned> scalarToAVX;
```

Comment:
I guess you can use `static const DenseMap<unsigned, unsigned> scalarToAVX  = { ... }`